python/grass/script/setup.py

```diff
@@ -24,17 +24,24 @@
     # query GRASS itself for its GISBASE
     # (with fixes for specific platforms)
     # needs to be edited by the user
-    grass8bin = 'grass'
-    if sys.platform.startswith('win'):
+    grass8bin = "grass"
+    if sys.platform.startswith("win"):
         # MS Windows
-        grass8bin = r'C:\OSGeo4W\bin\grass.bat'
+        grass8bin = r"C:\OSGeo4W\bin\grass.bat"
         # uncomment when using standalone WinGRASS installer
         # grass8bin = r'C:\Program Files (x86)\GRASS GIS 8.0.0\grass.bat'
         # this can be avoided if GRASS executable is added to PATH
-    elif sys.platform == 'darwin':
+    elif sys.platform == "darwin":
         # Mac OS X
-        # TODO: this have to be checked, maybe unix way is good enough
-        grass8bin = '/Applications/GRASS/GRASS-8.0.app/'
+        mac_gisbase = "/Applications/GRASS-8.0.app/Contents/Resources"
+        usr_module_dir = "~/Library/GRASS/8.0/Modules"
+        grass8bin = f"{mac_gisbase}/bin/grass"
+        os.environ["PATH"] = (
+            f"{mac_gisbase}/bin:{mac_gisbase}/scripts:"
+            f"{usr_module_dir}/bin:{usr_module_dir}/scripts:"
+            "/usr/bin:/bin:/usr/sbin:/etc:/usr/lib"
```

Comment:
There are [functions to do this](https://stackoverflow.com/a/1681244) in a better way, or is resetting the path your intention?

Reply:
The official binary `GRASS-8.0.app` type of program, is a contained conda environment, with the above set PATH on execution only (but not otherwise) . So yes, it was an intentional complete reset.

However, the example test works also with **only** setting `grass8bin`:

```python
elif sys.platform == "darwin":
    # Mac OS X
    grass8bin = "/Applications/GRASS-8.0.app/Contents/Resources/bin/grass"
```

Is setting PATH needed at all for this?

Reply:
The new logic is the following:

1. You call the binary to obtain the path needed to import the _grass_ package.
2. Leave the rest of the setup on the _grass.script.setup.init_ function.

With the these two notes:

1. On Windows and macOS, number 1 might be a challenge (unlike Linux where executable called _grass_ is assumed to be on path).
2. Ideally, the _grass_ package would be on Python path, so you would just need to get the right Python set up (NOP on standard Linux installation) instead of dealing with the executable in the script, i.e., ideally you would just `import grass` (#658).

So, if `/Applications/GRASS-8.0.app/Contents/Resources/bin/grass` runs GRASS GIS from command line on your machine, that's all what you (should) need to do before calling the _init_ function.

Reply:
Thanks for the explanation!